server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueMailDelete.java

```diff
@@ -59,22 +60,24 @@
     Mono<Void> considerDeleted(MailKey mailKey, MailQueueName mailQueueName) {
         return deletedMailsDao
             .markAsDeleted(mailQueueName, mailKey)
-            .doOnTerminate(() -> maybeUpdateBrowseStart(mailQueueName));
+            .doOnTerminate(() -> maybeUpdateBrowseStart(mailQueueName).subscribe());
     }
 
     Mono<Boolean> isDeleted(Mail mail, MailQueueName mailQueueName) {
         return deletedMailsDao.isDeleted(mailQueueName, MailKey.fromMail(mail));
     }
 
-    void updateBrowseStart(MailQueueName mailQueueName) {
```

Comment:
This looks quite synchronous to me. Upon termination this method will be called and will be blocking.

Reply:
It calls findNewBrowseStart that returns a Mono (so, async) and then updateNewBrowseStart that flatMap on this Mono (so it stays async). It doesn't look synchronous to me

Reply:
So in short that 'Mono' got evaluated because the source of the entire transformation is a COmpletableFuture?

Reply:
I think so

Reply:
That is the answer I was looking for... Things done from a future are hot sources I imagine